unified-runtime/source/adapters/level_zero/v2/command_buffer.hpp

```diff
@@ -51,11 +54,22 @@ struct ur_exp_command_buffer_handle_t_ : public ur_object {
       uint32_t numUpdateCommands,
       const ur_exp_command_buffer_update_kernel_launch_desc_t *updateCommands);
 
+  ur_exp_command_buffer_sync_point_t getSyncPoint(ur_event_handle_t event);
+  ur_event_handle_t *getWaitListFromSyncPoints(
+      const ur_exp_command_buffer_sync_point_t *pSyncPointWaitList,
+      uint32_t numSyncPointsInWaitList);
+
 private:
+  std::vector<ur_event_handle_t> syncPoints;
+
+  std::vector<ur_event_handle_t> syncPointWaitList;
```

Comment:
Are we using this member? I wouldn't think we'd need to keep a separate wait-list variant to the std::vector above

Reply:
This could use a comment (or maybe completely separate abstraction to store `syncPoints`).

 `syncPointWaitList` is for temporary contiguous event storage that we pass to command manager's appendXYZ API that requires a plain array of events.
Storing it in the class helps us avoid heap allocating this object every time.